plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
+	public static double calculateArea(Shape[] shapes) {
+		double area = 0.0;
+		if (shapes == null) {
+			return area;
```

Comment:
Then I suppose we would want to make the distinction between a null list
and an empty one. With the exception in place, the onus would then be on
the user to make sure their empty list of shapes was non-null (in my
example).

On Thu, Feb 5, 2015 at 4:18 PM, Sergey A. Razin <notifications@github.com>
wrote:

> In
> plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java
> <https://github.com/siostechcorp/exercises/pull/4#discussion_r24200402>:
>
> > @@ -0,0 +1,25 @@
> > +package planForExtensionExercise;
> > +
> > +public class TotalAreaCalculator extends AreaCalculator {
> > +	public static double calculateArea(Shape[] shapes) {
> > +		double area = 0.0;
> > +		if (shapes == null) {
> > +			return area;
>
> That's certainly valid, however could be misleading. Is area of a null 0?
> Maybe yes...what if you truly want the user to know that what was passed in
> is in fact an unexpected parameter (like null). What would you do without
> changing the return type, however (hint) it may require a method signature
> to be changed, and how?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siostechcorp/exercises/pull/4/files#r24200402>.
>